model/modeldecoder/rumv3/decoder.go

```diff
@@ -161,20 +161,22 @@ func DecodeNestedTransaction(d decoder.Decoder, input *modeldecoder.Input, batch
 	*batch = append(*batch, transaction)
 
 	for _, m := range root.Transaction.Metricsets {
-		metricset := input.Base
-		mapToMetricsetModel(&m, &metricset)
-		metricset.Metricset.Transaction.Name = transaction.Transaction.Name
-		metricset.Metricset.Transaction.Type = transaction.Transaction.Type
-		*batch = append(*batch, metricset)
+		event := input.Base
+		mapToMetricsetModel(&m, &event)
+		event.Processor = model.MetricsetProcessor
```

Comment:
It appears this `event.Processor = model.MetricsetProcessor` is also being set in `mapToMetricsetModel`? Is that intentional?

Reply:
good catch, I've removed these ones and consolidated on setting it in mapToFooModel